plugin/trino-mariadb/src/test/java/io/trino/plugin/mariadb/TestMariaDbLatestConnectorTest.java

```diff
@@ -36,4 +70,383 @@ protected SqlExecutor onRemoteDatabase()
     {
         return server::execute;
     }
+
+    @BeforeClass
+    public void setUpTables()
+    {
+        server.execute("ANALYZE TABLE tpch.nation PERSISTENT FOR ALL;");
+        server.execute("ANALYZE TABLE tpch.region PERSISTENT FOR ALL;");
+        server.execute("ANALYZE TABLE tpch.orders PERSISTENT FOR ALL;");
+        server.execute("ANALYZE TABLE tpch.customer PERSISTENT FOR ALL;");
+    }
+
+    @Override
+    public void testJoinPushdown()
+    {
+        PlanMatchPattern joinOverTableScans =
+                node(JoinNode.class,
+                        anyTree(node(TableScanNode.class)),
+                        anyTree(node(TableScanNode.class)));
+
+        PlanMatchPattern broadcastJoinOverTableScans =
+                node(JoinNode.class,
+                        node(TableScanNode.class),
+                        exchange(ExchangeNode.Scope.LOCAL,
+                                exchange(ExchangeNode.Scope.REMOTE, ExchangeNode.Type.REPLICATE,
+                                        node(TableScanNode.class))));
+
+        Session session = joinPushdownEnabled(getSession());
+
+        if (!hasBehavior(SUPPORTS_JOIN_PUSHDOWN)) {
```

Comment:
Please avoid using `hasBehavior()` in connector side since the connectors already know their behavior.